clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/RequestBasedMetadata.java

```diff
@@ -451,6 +451,8 @@ synchronized void updateCache(boolean onDemandRefresh) throws InterruptedExcepti
         helixGroupInfo.put(entry.getKey().toString(), entry.getValue());
       }
 
+      routingStrategy.updateHelixGroupInfo(helixGroupInfo);
```

Comment:
do we have to validate whether everything in readytoserver replicas is in here and atleast log and probably a metric so we can know if there are any inconsistencies?